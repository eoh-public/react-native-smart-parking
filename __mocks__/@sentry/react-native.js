export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
