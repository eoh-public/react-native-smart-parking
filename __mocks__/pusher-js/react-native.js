const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
