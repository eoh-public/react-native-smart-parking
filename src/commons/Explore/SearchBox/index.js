import React, { memo, useCallback } from 'react';
import { TextInput, View, StyleSheet, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { t } from 'i18n-js';

import { Colors } from '../../../configs';
import Theme from '../../../configs/Theme';
import Search from '../../../../assets/images/Explore/Search.svg';
import ArrowBack from '../../../../assets/images/Explore/ArrowBack.svg';

const SearchBox = memo(({ isBack, style, onFocus }) => {
  const { goBack } = useNavigation();
  const onBack = useCallback(() => {
    goBack();
  }, [goBack]);
  const onFocusInput = useCallback(() => {
    onFocus && onFocus();
  }, [onFocus]);
  return (
    <View style={[styles.searchBox, style]}>
      {isBack ? (
        <TouchableOpacity style={styles.backBtn} onPress={onBack}>
          <ArrowBack />
        </TouchableOpacity>
      ) : (
        <View>
          <Search />
        </View>
      )}
      <TextInput
        placeholder={t('text_search_location_placeholder')}
        placeholderTextColor={Colors.Gray7}
        style={Theme.textSearchInput}
        onFocus={onFocusInput}
      />
    </View>
  );
});

export default SearchBox;
const styles = StyleSheet.create({
  searchBox: {
    ...Theme.flexRow,
    paddingHorizontal: 16,
    backgroundColor: Colors.White,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    borderRadius: 5,
    marginTop: 8,
    height: 48,
  },
  backBtn: {
    padding: 12,
    marginHorizontal: -12,
  },
});
