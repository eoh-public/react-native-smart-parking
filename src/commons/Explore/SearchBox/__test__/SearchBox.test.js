import SearchBox from '../index';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer, { act } from 'react-test-renderer';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      goBack: mockedNavigate,
    }),
  };
});

describe('Test SearchBox', () => {
  let wrapper;
  test('SearchBox render', () => {
    act(() => {
      wrapper = renderer.create(<SearchBox isBack={true} />);
    });
    const instance = wrapper.root;
    const button = instance.findByType(TouchableOpacity);

    act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
    expect(mockedNavigate.mock.calls.length).toBe(1);
  });
});
