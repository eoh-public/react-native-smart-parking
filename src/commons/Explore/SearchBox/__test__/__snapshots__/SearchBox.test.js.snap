// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchBox SearchBox render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "marginTop": 8,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": -12,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <SvgMock />
  </View>
  <TextInput
    allowFontScaling={true}
    onFocus={[Function]}
    placeholder="Tìm kiếm địa điểm công cộng, cửa hàng, nhà hàng,..."
    placeholderTextColor="#8C8C8C"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 14,
        "height": "100%",
        "marginLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
